### ISSUE TITLE
std::vector push_back in lua

### ISSUE THREAD

**Issue by Static-electro** [vinniefalco/LuaBridge#110]
Hi
I have some strange behaviour using std::vector with LuaBridge
Let's say we have Vec2 struct in C++

``` c++
struct Vec2
{
    float x;
    float y;
};
```

And I need to manipulate a vector of these structs. So I register them as follows:

```
.beginClass<Vec2>("Vec2")
   .addConstructor<void (*) (float, float)>()
   .addData("x", &Vec2::x)
   .addData("y", &Vec2::y)
.endClass()
.beginClass<std::vector<Vec2> >("Vec2List")
     .addConstructor<void (*) ()>()
     .addFunction<void (std::vector<Vec2>::*) (const Vec2&)>("push", &std::vector<Vec2>::push_back)
.endClass()
```

That seems to work. Almost. Consider the lua code:

``` lua
local pts = Vec2List()
pts:push(Vec2(200, 1)) -- this point will be present in the object
pts:push(Vec2(1, 200)) -- and this one, too
road.points = pts -- road is another c++ class, that contains vector<Vec2>
road.points:push(Vec2(30, 20)) -- but I can't add points this way. The vector (road.points in this case) remains unchanged
```

What am I missing?
Thanks in advance

**Comment by crazybie** [comment]
the code 'road.points' will create a temp vector instance. this is because luarbridge treat all reference types as value types.
try use addProperty and return a pointer of points to lua. maybe like this:

//add to Road class.
VecList* get_points() const { return const_cast<VecList*>(&points); } 
void set_points(VecList* v){ points=*v; }

// then register it as a property
.beginClass<Road>()
    .addProperty("points", &Road::get_points, &Road::set_points)
    // others ....

**Comment by Static-electro** [comment]
Thanks for the clarification. I'll try to use addProperty.

**Comment by dmitry-t** [comment]
This should have been fixed by #189.
